### PR TITLE
Created Friendship association

### PR DESCRIPTION
**Pull request by Osong-Michael**
**Created the Friendship model**
- User can send a friend request to another user
- The other User can either accept or delete the friend request
1.  Friendship status is of 3 types:     
2.     1  for friends
3.     -1 for pending
4.     0  for not friends

### CODE REVIEW

**Review by Ispirett — CHANGES_REQUESTED** [review]
## Status changes required 

Great job @OA7  so far you have done weill.

There are just a few things to take care of. Please see below. :+1: 

- [ ]  When two users have become friends  user.friendships and user.inverse_friendships should be populated. Currently, one or the other is being populated.

**Comment by Osong-Michael** [comment]
The user who sent the request can only call the `friendships` method, only the friend the request was sent to can call the `inverse_friendships` method. This is because the **Friendships** table has a `user_id` column that represents the user sending the request and a `friend_id` column representing the user receiving the request, and also the foreign key to access the `inverse_friendships` method is the` friend_id `

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,6 +1,42 @@
...
+          <li> <%= link_to friend.name, friend, class: 'name-links' %> : 
+           <%= link_to 'Accept Friend Request', user_create_friend_path({friend_id: friend.id, user_id: current_user.id}), class: 'link-btn' %> &nbsp; 
+           <%= link_to 'Reject Friend Request', user_reject_friend_request_path({friend_id: friend.id, user_id: current_user.id}), class: 'link-btn2' %> 
+          </li>
+        </ul>
+        <br>
+      <% end %>  
+    </article>
+
+  <% end %>
```
Comment (euqueme) [inline app/views/users/show.html.erb]: - [ ] Place the if statement in a helper method that you call here and within the helper method render a partial where you put all the markup that is contained in the if statement since there should be no logic in views

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request reviewed by Maria Eugenia Queme P. (TSE)

### Hello @OA7 :balloon: 

### Thank you for working on the changes requested by the previous reviewer :medal_sports: your project still needs some work before it can be approved :construction: :construction_worker: 

#### Behavior

- [ ]  The app lets the current user send a friend request to another user that already send a friend request to the current user, let's hide the send friend request button and instead show the confirm or decline buttons in the ALL USERS view also let's add a proper unit test case to avoid the duplicate inverse pending friend request from being created in the database.

image

* [ ] There is a button to invite the user to friendship on users list page and on the single-user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- Currently, the button is only in the ALL USERS page put it also in the user's Show page

* [ ] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- Currently, the timeline is showing all the post by all the users and not only the current user and his or her friends

#### Views & Helpers

* [ ] There should be absolutely NO logic in views.

  > Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

- I left a message directly in your code where I spotted logic (only in the changes you've worked on) and a suggestion on where to place it.

[OPTIONAL]
- I strongly recommend you also remove the logic that came with the original project.

### That's it! :raised_hands: 

### Please :pray: do the changes :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

### Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Comment by Anna-Myzukina** [comment]
## Hi team ! Very nice project! :wave: 

# You did a great job, approved  :+1:

### Go and merge to development.


`Goodluck in your next Milestone.`
Happy coding  <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>
